src/map/Map.js

```diff
@@ -785,9 +785,11 @@ L.Map = L.Evented.extend({
 		    se = bounds.getSouthEast(),
 		    size = this.getSize().subtract(padding),
 		    boundsSize = this.project(se, zoom).subtract(this.project(nw, zoom)),
-		    snap = L.Browser.any3d ? this.options.zoomSnap : 1;
+		    snap = L.Browser.any3d ? this.options.zoomSnap : 1,
+		    scalex = size.x / boundsSize.x,
+		    scaley = size.y / boundsSize.y,
+		    scale = inside ? Math.max(scalex, scaley) : Math.min(scalex, scaley);

```

Comment:
This could also be written as `scale = Math.[ inside ? 'max' : 'min' ](size.x / boundsSize.x, size.y / boundsSize.y)`.

But I'd rather make the code a bit more readable, even if it means making it a few bytes bigger.